src/Rule/Assertion/Declaration/ShouldHaveOnlyOnePublicMethodNamed/ShouldHaveOnlyOnePublicMethodNamed.php

```diff
@@ -0,0 +1,41 @@
+<?php declare(strict_types=1);
+
+namespace PHPat\Rule\Assertion\Declaration\ShouldHaveOnlyOnePublicMethodNamed;
+
+use PHPat\Configuration;
+use PHPat\Rule\Assertion\Declaration\DeclarationAssertion;
+use PHPat\Rule\Assertion\Declaration\ValidationTrait;
+use PHPat\Statement\Builder\StatementBuilderFactory;
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Type\FileTypeMapper;
+
+abstract class ShouldHaveOnlyOnePublicMethodNamed extends DeclarationAssertion
+{
+    use ValidationTrait;
+
+    public function __construct(
+        StatementBuilderFactory $statementBuilderFactory,
+        Configuration $configuration,
+        ReflectionProvider $reflectionProvider,
+        FileTypeMapper $fileTypeMapper
+    ) {
+        parent::__construct(
+            __CLASS__,
+            $statementBuilderFactory,
+            $configuration,
+            $reflectionProvider,
+            $fileTypeMapper
+        );
+    }
+
+    protected function applyValidation(string $ruleName, ClassReflection $subject, bool $meetsDeclaration, array $tips, array $params = []): array
+    {
+        return $this->applyShould($ruleName, $subject, $meetsDeclaration, $tips, $params);
+    }
+
+    protected function getMessage(string $ruleName, string $subject, array $params = []): string
+    {
+        return $this->prepareMessage($ruleName, sprintf('%s should have only one public methods named', $subject));
```

Comment:
```suggestion
        return $this->prepareMessage($ruleName, sprintf('%s should have only one public method named %s', $subject, $params['name']));
```

Reply:
And this should affect the test, that should be adapted to the correct error message